app/datasources/db/models.py

```diff
@@ -1,9 +1,64 @@
-from typing import Optional
+from sqlmodel import JSON, Column, Field, SQLModel, UniqueConstraint, select
 
-from sqlmodel import Field, SQLModel
 
+class SqlQueryBase:
 
-class Contract(SQLModel, table=True):
-    address: bytes = Field(nullable=False, primary_key=True)
+    @classmethod
+    async def get_all(cls, session):
+        result = await session.exec(select(cls))
+        return result.all()
+
+    async def _save(self, session):
+        session.add(self)
+        await session.commit()
+        return self
+
+    async def update(self, session):
+        return await self._save(session)
+
+    async def create(self, session):
+        return await self._save(session)
+
+
+class AbiSource(SqlQueryBase, SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
+    name: str = Field(nullable=False)
+    url: str = Field(nullable=False)
+
+
+class Abi(SqlQueryBase, SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
+    abi_hash: bytes = Field(nullable=False, index=True, unique=True)
+    relevance: int | None = Field(nullable=False, default=0)
+    abi_json: dict = Field(default_factory=dict, sa_column=Column(JSON))
+    source_id: int | None = Field(
+        nullable=True, default=None, foreign_key="abisource.id"
+    )
+
+
+class Project(SqlQueryBase, SQLModel, table=True):
+    id: int | None = Field(default=None, primary_key=True)
+    description: str = Field(nullable=False)
+    logo_file: str = Field(nullable=False)
+
+
+class Contract(SqlQueryBase, SQLModel, table=True):
+    __table_args__ = (
+        UniqueConstraint("address", "chain_id", name="address_chain_unique"),
+    )
+
+    id: int | None = Field(default=None, primary_key=True)
+    address: bytes = Field(nullable=False, index=True)
     name: str = Field(nullable=False)
-    description: Optional[str] = None
+    display_name: str | None = None
+    description: str | None = None
```

Comment:
No fields?

Reply:
As fas as I know Field is no needed in the basic case, nullable and default null. https://sqlmodel.tiangolo.com/tutorial/create-db-and-table/#optional-fields-nullable-columns